lib/ansible/module_utils/basic.py

```diff
@@ -1665,7 +1670,10 @@ def _check_argument_types(self):
             try:
                 self.params[k] = type_checker(value)
             except (TypeError, ValueError):
-                self.fail_json(msg="argument %s is of type %s and we were unable to convert to %s" % (k, type(value), wanted))
+                e = get_exception()
+                self.fail_json(
+                    msg="argument %s is of type %s and we were unable to convert to %s" % (k, type(value), wanted),
+                    details=str(e))
```

Comment:
use 'exception' key which is already there for this kind of thing, 'details' is not something ansible recognizes.